nautobot/core/factory.py

```diff
@@ -147,9 +162,25 @@ def get_random_instances(model_or_queryset_or_lambda, minimum=0, maximum=None):
     )
 
 
-class UniqueFaker(factory.Faker):
-    """https://github.com/FactoryBoy/factory_boy/pull/820#issuecomment-1004802669"""
+class NautobotBoolIterator(factory.Iterator):
+    """Factory iterator that returns a semi-random sampling of boolean values
 
-    @classmethod
-    def _get_faker(cls, locale=None):
-        return super()._get_faker(locale=locale).unique
+    Iterator that returns a random sampling of 4 True and 4 False values to guarantee a
+    50/50 distribution of values. Used in factories when a data set must contain both
+    True and False values.
+    """
+
+    def _nautobot_boolean_iterator_sample(self):
+        iterator = [True, True, True, True, False, False, False, False]
+        factory.random.randgen.shuffle(iterator)
+        return iterator
+
+    def __init__(self, *args, cycle=True, getter=None):
```

Comment:
not blocking for this PR, but it'd be sweet to add support for a [`chance_of_getting_true`](https://faker.readthedocs.io/en/stable/providers/faker.providers.misc.html#faker.providers.misc.Provider.boolean) parameter that could be used to weight the scales even further. :-)

Reply:
Do you think it would be useful to be able to explicitly set the number of `True` and `False` values in the list?

Reply:
Yeah, that could be another way to achieve the same result. Good thought. I was thinking something probably dumb like:

```python
iterator = [(i < chance_of_getting_true) for i in range(0, 100)]
```

Reply:
https://github.com/nautobot/nautobot/pull/3211/commits/91f60cf8e5281d976c870aa3df5e200cf3294b1b

Reply:
Now I'm thinking it needs tests

Reply:
yes please! :-) No good deed, etc. etc.